dvc/repo/experiments/init.py

```diff
@@ -159,32 +159,24 @@ def validate_prompts(
 ) -> Union[Any, Tuple[Any, str]]:
     from dvc.ui.prompt import InvalidResponse
 
+    msg_format = "[yellow]'{0}' does not exist, " "the {1} will be created. "
     if key == "params":
-        import errno
-
         from dvc.dependency.param import ParamsDependency
 
         assert isinstance(value, str)
-        msg_format = (
-            "[prompt.invalid]'{0}' {1}. "
-            "Please retry with an existing parameters file."
-        )
         try:
             ParamsDependency(None, value, repo=repo).validate_filepath()
-        except (IsADirectoryError, FileNotFoundError) as e:
-            suffices = {
-                errno.EISDIR: "is a directory",
-                errno.ENOENT: "does not exist",
-            }
-            raise InvalidResponse(msg_format.format(value, suffices[e.errno]))
+        except FileNotFoundError:
+            return value, msg_format.format(value, "file")
+        except IsADirectoryError:
+            raise InvalidResponse(
+                f"[prompt.invalid]'{value}' is a directory. "
+                "Please retry with an existing parameters file."
+            )
     elif key in ("code", "data"):
         if not os.path.exists(value):
             typ = "file" if is_file(value) else "directory"
-            return (
-                value,
-                f"[yellow]'{value}' does not exist, "
-                f"the {typ} will be created. ",
-            )
+            return value, msg_format.format(value, typ)
```

Comment:
Actually, I didn't like this design, sometimes this function returns a tuple and sometimes it only returns one value. It will make it hard to use.

Reply:
It won't be used directly, this is a callback function, so the "Prompt" handles this automatically. The other option would be to do `return value, None` which is also iffy.